models/Document.php

```diff
@@ -807,7 +807,7 @@ public function getPath(): ?string
     {
         // check for site, if so rewrite the path for output
         try {
-            if (Tool::isFrontend() && Site::isSiteRequest()) {
+            if ($this->path != null && Tool::isFrontend() && Site::isSiteRequest()) {
```

Comment:
Thank you for the PR

I'd like to suggest
```suggestion
            if ($this->path && Tool::isFrontend() && Site::isSiteRequest()) {
```
so it all skips all the Sites related stuff when path is an empty string